mantidproject/mantid
# Stitch1DMany refactoring

Description of work.

The main aim of this refactoring was to enable the `ManualScaleFactors` property to take a list of multiple values instead of just one. During stitching, each of these scale factors can be applied separately to each input workspace. In addition a significant amount of other refactoring was done to simplify the code as it was getting very complicated.

Edit: Another new thing this refactor is forcibly ensure that `InputWorkspaces` can only take a list of MatrixWorkspaces or WorkspaceGroups containing MatrixWorkspaces only.

**To test:**

- Mostly code review
- Ensure unit tests pass and the newly added ones make sense
- Ensure the relevant documentation is updated and makes sense



Fixes #18378.

*Release notes will be updated later.*

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
